### PR TITLE
QML Controls

### PR DESCRIPTION
**Pull request by jusko**
The aim here is to build on top of @Be-ing's [initial POC](https://github.com/mixxxdj/mixxx/pull/3448) and get a lightweight UI implemented.

- [x] Refactor the POC slider into a component
- [ ] Refactor `ControlProxyQML` to work directly with `ControlObject`
- [ ] Add the slider into a `Mixxx` namespace

Possibly in a new PR:
- [ ] Create a knob component
- [ ] Build a minimal UI with these two components

### CODE REVIEW

File: res/qml/main.qml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    width: 34
+    height: 100
+    visible: true
+
+    MixxxSlider {
+        width: window.width
+        height: window.height
+        group: "[Channel1]"
+        key: "volume"
+    }
```
Comment (Be-ing) [inline res/qml/main.qml:15]: Nice, this is very concise.

**Comment by Be-ing** [comment]
As a next step, I suggest make a similar custom Dial component with an arc like LateNight PaleMoon.

File: res/qml/MixxxSlider.qml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        width: parent.width
+        height: parent.height
+
+        Control {
+            id: control
+            parameter: slider.value
+        }
+
+        handle: Image {
+            source: "../skins/LateNight/classic/sliders/knob_volume_deck.svg"
```
Comment (Be-ing) [inline res/qml/MixxxSlider.qml:24]: Use generically named variables for the paths to the graphics assets ("slider-background", "slider-handle"). https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/QML/near/136221715

**Comment by Be-ing** [comment]
I wonder if we can apply a rotation transformation to the SVGs for a horizontal slider without needing separate SVGs.

**Comment by ronso0** [comment]
> As a next step, I suggest making a similar custom Dial component with an arc like LateNight PaleMoon.

Let's see how a simple dial works first as the ring requires some extra parameters and code.
If the simple goes well

> I wonder if we can apply a rotation transformation to the SVGs for a horizontal slider without needing separate SVGs.

this is included, right.

**Comment by daschuer** [comment]
Thank you for adopting this work. I am very happy to see progress.

I am a bit concerned about the length of your to-do list. Please let us go in small mergable steps.
Else there is a risk of creating an un-mergable beast no one is willing to take the responsibility for pressing merge. See our PR backlog 

> Would be interested to know a bit more about when one would want to use ControlProxy instead @daschuer? In controlproxy.h it states:

>>  It should be used for new code to avoid unnecessary locking during send if no slot is connected.

> Doesn't this apply to QML? Or should ControlProxyQML be changed to use ControlObject directly (inlining the proxy functions as you say), but also ensure to connect to all CO's slots?

Yes, it does also apply to ControlProxyQML. An that is actually the issue in the current implementation. The ControlProxyQML always connects, so the original ControlProxy approach to avoid unesseary locking is bypassed.

I think it would be the best to reimplemeny ControlProxyXML to connect directly to the Control object. I don't mind if we duplicate some code in the first place. This helps to have a clear view what the object does.

**Comment by jusko** [comment]
Just to say that I'll be trying hard to get back into some daily dev on this soon. Been distracted by fighting with the bank who closed my business account in error

**Comment by Holzhaus** [comment]
Resolved merge conflicts with main here: https://github.com/Holzhaus/mixxx/tree/qml

But i'm getting a bunch of debug assertions due to non-existing control objects.

**Comment by Holzhaus** [comment]
Superseded by #3894. 

@jusko I cherry picked some stuff from this branch and integrated it into #3894. In addition to the slider, I also added a six implementations of knobs and buttons. Feel free to improve the demo skin further (more/better buttons, VU meters, etc.).

**Pull request closed without merging**